core/src/main/scala/org/bitcoins/core/protocol/dlc/models/DLCPayoutCurve.scala

```diff
@@ -83,29 +71,66 @@ case class DLCPayoutCurve(points: Vector[OutcomePayoutPoint]) {
       rounding: RoundingIntervals,
       totalCollateral: Satoshis): Satoshis =
     getPayout(outcome, rounding, totalCollateral)
+
+  def flip(totalCollateral: Satoshis): DLCPayoutCurve = {
+    DLCPayoutCurve(pieces.map(_.flip(totalCollateral)))
+  }
 }
 
-object DLCPayoutCurve {
+object DLCPayoutCurve
+    extends TLVDeserializable[PayoutFunctionV0TLV, DLCPayoutCurve](
+      PayoutFunctionV0TLV) {
+
+  override def fromTLV(tlv: PayoutFunctionV0TLV): DLCPayoutCurve = {
+    val pieces =
+      tlv.endpoints.init.zip(tlv.endpoints.tail).zip(tlv.pieces).map {
+        case ((leftEndpoint, rightEndpoint), tlvPiece) =>
+          DLCPayoutCurvePiece.fromTLV(leftEndpoint, tlvPiece, rightEndpoint)
+      }
+
+    DLCPayoutCurve(pieces)
+  }
+
+  def polynomialInterpolate(
+      points: Vector[PiecewisePolynomialPoint]): DLCPayoutCurve = {
+    require(points.head.isEndpoint && points.last.isEndpoint,
+            s"First and last points must be endpoints: $points")
+
+    val (_, _, pieces) = points.tail.foldLeft(
+      (points.head,
+       Vector.empty[PiecewisePolynomialMidpoint],
+       Vector.empty[DLCPolynomialPayoutCurvePiece])) {
+      case ((lastEndpoint, midpointsSoFar, piecesSoFar), point) =>
+        point match {
+          case midpoint: PiecewisePolynomialMidpoint =>
+            (lastEndpoint, midpointsSoFar.:+(midpoint), piecesSoFar)
+          case endpoint: PiecewisePolynomialEndpoint =>
+            val points = midpointsSoFar
+              .+:(lastEndpoint)
+              .:+(endpoint)
+              .map(_.toOutcomePayoutPoint)
+
+            (endpoint,
+             Vector.empty,
+             piecesSoFar.:+(DLCPolynomialPayoutCurvePiece(points)))
+        }
+    }
+
+    DLCPayoutCurve(pieces)
+  }
 
-  def fromTLV(tlv: PayoutFunctionV0TLV): DLCPayoutCurve = {
-    DLCPayoutCurve(tlv.points.map { point =>
-      val payoutWithPrecision =
-        point.value.toLong + (BigDecimal(point.extraPrecision) / (1 << 16))
-      OutcomePayoutPoint(point.outcome, payoutWithPrecision, point.isEndpoint)
-    })
+  def fromPoints(points: Vector[TLVPoint]): DLCPayoutCurve = {
+    //i don't think this is right
+    val curvePiece = DLCPolynomialPayoutCurvePiece(
+      points.map(_.outcomePayoutPoint))
+    val payoutCurve = DLCPayoutCurve(Vector(curvePiece))
+    payoutCurve
```

Comment:
What is the intention here? I think you mean to be using `polynomialInterpolate` or something like that? That is the only situation where points are enough to describe a function?

Reply:
As is, this is definitely wrong as you're putting all the curves into one piece so it is interpolating with some crazy high degree polynomial, instead of straight lines like I assume you want.

Reply:
I want to take a set of points and construct a curve out of it.

Reply:
It's not clear to me how to translate a `Vector[TLVPoint]` into `Vector[PiecewisePolynomialPoint]` that is required for `DLCPayoutCurve.polynomialInterpolate()`. Do i assume that these are all just `PiecewisePolynomialEndpoint` ?